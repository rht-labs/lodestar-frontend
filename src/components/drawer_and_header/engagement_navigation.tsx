import React from 'react';
import { Nav, NavExpandable, NavItem, NavList } from '@patternfly/react-core';
import { Link, useLocation } from 'react-router-dom';

export const EngagementNavigation = () => {
  const { pathname } = useLocation();

  return (
    <div>
      <Nav theme="dark">
        <NavList>
          <NavExpandable
            title={'Engagements'}
            isActive={pathname.includes('/app/engagements')}
            isExpanded
            id="navExpandableEngagement"
          >
            <NavItem
              id="Overview"
              itemId={0}
              isActive={pathname === '/app/engagements/overview'}
            >
              <Link to="/app/engagements/all">Overview</Link>
            </NavItem>
            <NavItem
              id="upcoming"
              itemId={0}
              isActive={pathname === '/app/engagements/upcoming'}
            >
              <Link to="/app/engagements/upcoming">Upcoming</Link>
            </NavItem>
            <NavItem
              id="Active"
              itemId={0}
              isActive={pathname === '/app/engagements/active'}
            >
              <Link to="/app/engagements/active">Active</Link>
            </NavItem>
            <NavItem
              id="Past"
              itemId={0}
              isActive={pathname === '/app/engagements/past'}
            >
              <Link to="/app/engagements/past">Past</Link>
            </NavItem>
            <hr className="pf-c-divider" />
            <NavItem
              id="New"
              itemId={0}
              isActive={pathname === '/app/engagements/new'}
            >
              <Link to="/app/engagements/new">+ Create New</Link>
            </NavItem>
          </NavExpandable>
        </NavList>
      </Nav>
    </div>
  );
};
