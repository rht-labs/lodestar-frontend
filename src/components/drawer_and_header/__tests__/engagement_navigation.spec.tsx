import React from 'react';
import {render, cleanup} from '@testing-library/react';
import {EngagementNavigation} from '../engagement_navigation';
import {MemoryRouter} from 'react-router';

afterEach(cleanup);

describe('Engagement Navigation ', () => {
  test('matches the snapshot', () => {
    expect(
      render(
        <MemoryRouter>
          <EngagementNavigation/>
        </MemoryRouter>
      )
    ).toMatchSnapshot();
  });
});
