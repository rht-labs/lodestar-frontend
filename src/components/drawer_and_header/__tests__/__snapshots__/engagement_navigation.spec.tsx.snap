// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Engagement Navigation  matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <nav
          aria-label="Global"
          class="pf-c-nav"
          data-ouia-component-id="1"
          data-ouia-component-type="PF4/Nav"
          data-ouia-safe="true"
        >
          <ul
            class="pf-c-nav__list"
          >
            <li
              class="pf-c-nav__item pf-m-expanded"
            >
              <a
                aria-expanded="true"
                class="pf-c-nav__link"
                href="#"
                id="navExpandableEngagement"
              >
                Engagements
                <span
                  class="pf-c-nav__toggle"
                >
                  <span
                    class="pf-c-nav__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                        transform=""
                      />
                    </svg>
                  </span>
                </span>
              </a>
              <section
                aria-labelledby="navExpandableEngagement"
                class="pf-c-nav__subnav"
              >
                
                <ul
                  class="pf-c-nav__list"
                >
                  <li
                    class="pf-c-nav__item"
                  >
                    <a
                      class="pf-c-nav__link"
                      href="/app/engagements/all"
                    >
                      Overview
                    </a>
                  </li>
                  <li
                    class="pf-c-nav__item"
                  >
                    <a
                      class="pf-c-nav__link"
                      href="/app/engagements/upcoming"
                    >
                      Upcoming
                    </a>
                  </li>
                  <li
                    class="pf-c-nav__item"
                  >
                    <a
                      class="pf-c-nav__link"
                      href="/app/engagements/active"
                    >
                      Active
                    </a>
                  </li>
                  <li
                    class="pf-c-nav__item"
                  >
                    <a
                      class="pf-c-nav__link"
                      href="/app/engagements/past"
                    >
                      Past
                    </a>
                  </li>
                  <hr
                    class="pf-c-divider"
                  />
                  <li
                    class="pf-c-nav__item"
                  >
                    <a
                      class="pf-c-nav__link"
                      href="/app/engagements/new"
                    >
                      + Create New
                    </a>
                  </li>
                </ul>
              </section>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <nav
        aria-label="Global"
        class="pf-c-nav"
        data-ouia-component-id="1"
        data-ouia-component-type="PF4/Nav"
        data-ouia-safe="true"
      >
        <ul
          class="pf-c-nav__list"
        >
          <li
            class="pf-c-nav__item pf-m-expanded"
          >
            <a
              aria-expanded="true"
              class="pf-c-nav__link"
              href="#"
              id="navExpandableEngagement"
            >
              Engagements
              <span
                class="pf-c-nav__toggle"
              >
                <span
                  class="pf-c-nav__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                      transform=""
                    />
                  </svg>
                </span>
              </span>
            </a>
            <section
              aria-labelledby="navExpandableEngagement"
              class="pf-c-nav__subnav"
            >
              
              <ul
                class="pf-c-nav__list"
              >
                <li
                  class="pf-c-nav__item"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/app/engagements/all"
                  >
                    Overview
                  </a>
                </li>
                <li
                  class="pf-c-nav__item"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/app/engagements/upcoming"
                  >
                    Upcoming
                  </a>
                </li>
                <li
                  class="pf-c-nav__item"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/app/engagements/active"
                  >
                    Active
                  </a>
                </li>
                <li
                  class="pf-c-nav__item"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/app/engagements/past"
                  >
                    Past
                  </a>
                </li>
                <hr
                  class="pf-c-divider"
                />
                <li
                  class="pf-c-nav__item"
                >
                  <a
                    class="pf-c-nav__link"
                    href="/app/engagements/new"
                  >
                    + Create New
                  </a>
                </li>
              </ul>
            </section>
          </li>
        </ul>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
